shadowsocks-csharp/Controller/Service/UpdateChecker.cs

```diff
@@ -1,254 +1,174 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
 using System.Net;
+using System.Net.Http;
 using System.Text.RegularExpressions;
-
+using System.Threading.Tasks;
+using System.Windows;
 using Newtonsoft.Json.Linq;
 using NLog;
+using Shadowsocks.Localization;
 using Shadowsocks.Model;
 using Shadowsocks.Util;
+using Shadowsocks.Views;
 
 namespace Shadowsocks.Controller
 {
     public class UpdateChecker
     {
-        private static Logger logger = LogManager.GetCurrentClassLogger();
+        private readonly Logger logger;

```

Comment:
Why convert to member variables?

Reply:
Because it should've been a read-only private field in the first place. The `UpdateChecker` class has only one instance managed by `MenuViewController` throughout the lifecycle. There's no need to have any static fields here. Having unnecessary static fields breaks the code hierarchy.